adapters/latest/src/main/java/org/apache/cassandra/sidecar/adapters/latest/CassandraStorageOperations.java

```diff
@@ -64,8 +73,28 @@ public void takeSnapshot(@NotNull String tag, @NotNull String keyspace, @NotNull
         requireNonNull(tag, "snapshot tag must be non-null");
         requireNonNull(keyspace, "keyspace for the  must be non-null");
         requireNonNull(table, "table must be non-null");
-        jmxClient.proxy(StorageJmxOperations.class, STORAGE_SERVICE_OBJ_NAME)
-                 .takeSnapshot(tag, options, keyspace + "." + table);
+        try
+        {
+            jmxClient.proxy(StorageJmxOperations.class, STORAGE_SERVICE_OBJ_NAME)
+                     .takeSnapshot(tag, options, keyspace + "." + table);
+        }
+        catch (IOException e)
+        {
+            if (StringUtils.contains(e.getMessage(),
+                                     "Snapshot " + tag + " already exists"))
+            {
+                throw new SnapshotAlreadyExistsException(e.getMessage());
+            }
+            else if (StringUtils.contains(e.getMessage(), "Keyspace " + keyspace + " does not exist"))
+            {
+                throw new IllegalArgumentException(e.getMessage());
+            }
+            else if (StringUtils.contains(e.getMessage(),
+                                          "Cannot snapshot until bootstrap completes"))
+            {
+                throw new NodeBootstrappingException(e.getMessage());
+            }
```

Comment:
you are swallowing any other exception error message. We should propagate it anyway
```suggestion
            }
            throw new Runtime(e);
```